### PR TITLE
✨ Fallback Tool for REDdeploy Plus Settings Logic Unification

### PR DESCRIPTION
**Pull request by effs**
Four main things here:

- [x] The Tool, which is what I set out to do. It literally just runs `redMod.exe deploy` so it's not very useful (unless you have no conflicts or manage those with alpha)

- [x] After a very reasonable existential crisis, put the store get/set behind a pair of functions and tried to make sure that everything else is either using those or Features where needed. Most of the logic refactoring is to support this and it's ugly af.

- [x] Tweaked copy

- [x] Deleted currently unused stuff that we may want to reintroduce

### CODE REVIEW

**Review by E1337Kat — APPROVED** [review]

**Comment by E1337Kat** [comment]
oh... I thought I figured out how to approve an individual commit for once. nm that... for now.

File: src/reducers.ts [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
   reducers: {
-    [actions.setArchiveAutoConvert as any]: (state, payload) =>
-      vortexApi.util.setSafe(state, [`v2077_feature_redmod_autoconvert_archives`], payload),
-    [actions.setRedmodForceDeploy as any]: (state, payload) =>
-      vortexApi.util.setSafe(state, [`redmodForceDeploy`, payload.profileId], payload.force),
-  },
-  defaults: {
-    archiveAutoConvert: false,
-    redmodForceDeploy: {},
+    [actions.setREDmodAutoconvertArchivesAction.toString()]: (state, payload: boolean) =>
```
Comment (E1337Kat) [inline src/reducers.ts:14]: oh, you found it is a string. I didn't know where to look to find that info. :/

File: src/redmodding.ts [hunk 2]
```diff
@@ -36,11 +37,13 @@ import {
...
   VortexDiscoveryResult,
   VortexExtensionContext,
+  VortexIToolShim,
   VortexState,
   VortexToolDiscovered,
 } from "./vortex-wrapper";
 
-// This function runs on starting up Vortex or switching to Cyberpunk as the active game. This may need to be converted to a test, but the UI for tests is less flexible.
+// This function runs on starting up Vortex or switching to Cyberpunk as the active game.
+// This may need to be converted to a test, but the UI for tests is less flexible.
```
Comment (E1337Kat) [inline src/redmodding.ts:46]: I feel like this comment is misplaced...

Comment (E1337Kat) [reply src/redmodding.ts:46]: Based on the original file: https://github.com/E1337Kat/cyberpunk2077_ext_redux/blob/d7c7dd93b1266f4cc8f628b5eecbccfccf85203a/src/prepareForModding.ts, I believe the comment referred to the default exported function prepareForModding, which is now the wrapped `wrappedPrepareForModdingWithREDmodding` or the "internal" `prepareForModdingWithREDmodding` I thing adding the comment as the comment doc to the wrapped function would be a good change to make it obvious what is going on here.

Comment (E1337Kat) [reply src/redmodding.ts:46]: (I know that isn't a part of this PR, but this change made me think of it)

File: src/redmodding.ts [hunk 3]
```diff
@@ -128,10 +129,18 @@ const autoDeployActionCondition: V2077ActionConditionFunc = (
...
 };
 
+export const internalSettingsViewStuff: V2077SettingsView = {
+  type: ActionType.AutoRunDeploy,
+  id: ``,
+  actionOrCondition: autoDeployAction,
+  condition: autoDeployActionCondition,
+  // test: checkTest,
+};
+
```
Comment (E1337Kat) [inline src/redmodding.ts:143]: I know I added this, but the name leaves something to be desired. Also I am not sure it is accurate. It's more like this deals with the ToDo side of the settings, not the Settings View really. Don't bother changing it in this commit though. I will get it taken care of myself if a refactoring ticket. Just kinda noting out my thoughts as I go.

**Review by E1337Kat — COMMENTED** [review]
Mostly good so far. just a few thoughts thus far.

File: src/reducers.ts (hunk 1 above)
Comment (effs) [reply src/reducers.ts]: Yeah, ran into it somewhere in the Redux docs

File: src/redmodding.ts (hunk 3 above)
Comment (effs) [reply src/redmodding.ts]: I nixed all this for now - I think it looks like something we may want to reintroduce but I tried to leave only code in use ATM

**Review by E1337Kat — APPROVED** [review]
seems pretty good.

**Comment by E1337Kat** [comment]
Granted, this approval comes when I am very very tired and sleepy, have a headache, and at 2am... so mayyyyybe make me look again?? :p

**Pull request merged**